### PR TITLE
Chore: Update Prometheus

### PR DESCRIPTION
**Pull request by alanprot**
**What this PR does**:
Update Prometheus to https://github.com/prometheus/prometheus/commit/d81e41d58ec89c3e55713a03a9c4a3fbc56eaf0e and thanos to https://github.com/thanos-io/thanos/commit/943401f726ed06cba9b749fc7533023f5fea58f4

**THIS UPDATE DEPRECATE/REMOVE ALL V1 AM APIs following https://github.com/prometheus/alertmanager/pull/2970**

Relevant Changes:
* https://github.com/prometheus/alertmanager/pull/2970
* https://github.com/prometheus/prometheus/pull/13461
* https://github.com/prometheus/alertmanager/pull/3045
* https://github.com/prometheus/alertmanager/pull/3483
* https://github.com/prometheus/alertmanager/pull/3567
* https://github.com/prometheus/alertmanager/pull/3604

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [NA] Tests updated
- [NA] Documentation added
- [NA] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/alertmanager/distributor.go [hunk 1]
```diff
@@ -78,21 +78,12 @@ func (d *Distributor) isUnaryDeletePath(p string) bool {
...
 	}
 	if strings.HasSuffix(p, "/v2/alerts/groups") {
 		return true, merger.V2AlertGroups{}
 	}
-	if strings.HasSuffix(p, "/v1/silences") {
-		return true, merger.V1Silences{}
-	}
-	if strings.HasSuffix(path.Dir(p), "/v1/silence") {
-		return true, merger.V1SilenceID{}
-	}
```
Comment (yeya24) [inline pkg/alertmanager/distributor.go:95]: Should we mention the v1 API deprecation in the changelog?

Comment (alanprot) [reply pkg/alertmanager/distributor.go:95]: Yeah.. i was just doing that! :D

**Comment by alanprot** [comment]
@yeya24 PTAL?

**Comment by yeya24** [comment]
Now what’s the behavior if we call v1 API? It is 404 or 410?

**Review by yeya24 — APPROVED** [review]
Thanks! Great work

**Pull request merged**